### PR TITLE
Refunds

### PR DESCRIPTION
**Pull request by thip**
Give change when people pay too much for a property

### CODE REVIEW

File: blockchain/contracts/PropertyOwnership.sol [hunk 1]
```diff
@@ -14,7 +14,9 @@ contract PropertyOwnership is PropertyBase, ERC721Token {
   function buy(uint256 _propertyId) payable {
     require(msg.value >= properties[_propertyId].buy_price);
 
-    //
+    if (msg.value > properties[_propertyId].buy_price) {
+      msg.sender.send(msg.value - (properties[_propertyId].buy_price * 10 ** 18));
```
Comment (StevenLeighton21) [inline blockchain/contracts/PropertyOwnership.sol]: Don't understand what this line is doing!

File: blockchain/contracts/PropertyOwnership.sol [hunk 2]
```diff
@@ -14,7 +14,9 @@ contract PropertyOwnership is PropertyBase, ERC721Token {
   function buy(uint256 _propertyId) payable {
     require(msg.value >= properties[_propertyId].buy_price);
 
-    //
+    if (msg.value > properties[_propertyId].buy_price) {
```
Comment (thip) [inline blockchain/contracts/PropertyOwnership.sol]: property buy price should be converted to wei here too!

File: blockchain/contracts/PropertyOwnership.sol (hunk 1 above)
Comment (thip) [reply blockchain/contracts/PropertyOwnership.sol]: It's converting the buy_price from ether to wei - we should pull it out into a variable, especially seeing as it should be being used on line 17 too.

Comment (naliwajek) [reply blockchain/contracts/PropertyOwnership.sol]: Can we extract it to a `pure` func?

**Review by naliwajek — APPROVED** [review]

File: blockchain/contracts/PropertyOwnership.sol (hunk 2 above)
Comment (emileswarts) [reply blockchain/contracts/PropertyOwnership.sol]: whoa, how is this even passing the tests

**Pull request merged**